src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -101,13 +325,115 @@ def algorithm(self, parameters=None):
+
+            # find the pileup files
+            puFiles = findFiles(sandboxDir, "pileupconf.json")
+            self.logger.debug("### puFiles %s", puFiles)
+
+            # json dictionary to write back to our tarball, it has the following structure
+            # {pileupConfFile: pileupConfContent, ...}
+            # for example
+            # {'relative/path/pileupconf.json': {...}}
+            jdict = {}
+
+            # relative path name to our pileup configuration json file
+            puConfJson = ""
+            # list over each pileupconf.json and update it
+            for puFile in puFiles:
+                # load the JSON and figure out the dataset name
+                with open(puFile, 'r', encoding='utf-8') as istream:
+                    # puJsonContent has the following data-structure:
+                    # {<dtype>: {<block_name>: {"FileList": [list_of_files]}}}
+                    # for example
+                    # {'mc': {'/path/block#123': {"FileList: ["/file1.root", "/file2.root"]}}}
+                    puJsonContent = json.load(istream)
+
+                    # our puFile is path to pileupconf.json, we need to construct
+                    # relative path within our tarball,
+                    # therefore we only take WMSandbox/.../pileupconf.json part
+                    puConfJson = os.path.join('WMSandbox', puFile.split('WMSandbox/')[-1])
+
+                jsonPUName = ""
+                for _puType, blocks in puJsonContent.items():
+                    for blockName, _content in blocks.items():
+                        jsonPUName = blockName.split("#")[0]
+                        break  # we already have the pileup name
+                self.logger.info("Found pileup name %s under path: %s", jsonPUName, puFile)
+                self.logger.debug("### Found pileup name %s under path: %s", jsonPUName, puFile)
+
+                # now that we know the pileup name, iterate over the MSPileup docs
+                for pileupDoc in msPileupList:
+                    # check if active pileup workflow is found in MSPileup one
+                    pileupName = pileupDoc['pileupName']
+                    if pileupName == jsonPUName:
+                        # then we need to check whether there are any changes or not
+                        jsonBlockLoc = blockLocations(puJsonContent)
+
+                        # construct new data-structure:
+                        # - dict of blocks and rses mapping, it will be used by checkChanges
+                        msPUBlockLoc = {}
+                        for blk in pileupDoc["blocks"]:
+                            msPUBlockLoc[blk] = pileupDoc["rses"]
+
+                        # are the block locations different between the JSON and MSPileup?
+                        # jsonBlockLoc and msPUBlockLoc have identical data-structures
+                        # {block1: [rses], block2: [rses], ...}
+                        if checkChanges(jsonBlockLoc, msPUBlockLoc):
```

Comment:
to do that I suggest to put these messages into `updateBlockInfo` function. But since we use `logger` and not `print` which solution do you prefer:
1. add `logger` as argument to `updateBlockInfo` function and use it over there (my preference)
2. make `updateBlockInfo` as member function of `WorkflowUpdaterPoller` class which will have `self.logger` and therefore it can be used
I'm in favor of 1 due to easy testing, i.e. we don't need a class initialization to test this function.

Reply:
I would go with 1. as well. Yes, this function is probably better to remain a function instead of a class method.

Reply:
done